Load environment variables for deploy image

milesgranger opened the pull request:
Load environment variable(s): `DEPLOY_IMAGE` which should be set in a config map

milesgranger committed:
Add config map support

Specifically, to address change the deploy image via k8s
config map.

epa095 commented on src/main.rs:
```diff
@@ -22,6 +25,30 @@ pub struct GordoSpec {
+}
+impl From<BTreeMap<String, String>> for GordoControllerConfigMap {
+    fn from(data: BTreeMap<String, String>) -> Self {
+        let default_config = GordoControllerConfigMap::default();
```
I think the usual way is to let the deployment mount in the values from the configmap as enviroment variables as in this example: https://blog.bigbinary.com/2017/05/25/using-kubernetes-configmap-with-configuration-files-for-deploying-rails-app.html  Then 
1. The program dont need to know about the configmap at all, and 
2. It is visible from `kubectl describe deployment ...` what configuration is used.

milesgranger committed:
Switch to reflector pattern for monitoring config map changes

milesgranger committed:
Use environment variables for config instead of configmap

milesgranger committed:
Rename config to kube_config

milesgranger committed:
Panic when failing to poll

k8s will restart the controller, perhaps to avoid 'too old resource version' issue

epa095 reviewed (commented):
LGTM
